### DISCUSSION TITLE
How to customize the table name of an relationship table in an M2M relationship?

### DISCUSSION THREAD

**Discussion by CyberXNomad** [ent/ent#4263] [Q&A]
How to customize the table name of an relationship table in an M2M relationship?

**Comment by hexvalid** [comment]
Set the storage key on the owning `edge.To` using `edge.Table`:

```go
func (User) Edges() []ent.Edge {
    return []ent.Edge{
        edge.To("groups", Group.Type).
            StorageKey(edge.Table("membership_links")),
    }
}
```

The schema imports are `entgo.io/ent` and `entgo.io/ent/schema/edge`. If you also declare the inverse edge on `Group`, keep it as:

```go
edge.From("users", User.Type).Ref("groups")
```

That reference uses the storage configuration of the owning edge; the join table is `membership_links`. Regenerate the Ent code and apply the corresponding database migration.

I verified this with Ent v0.14.6: generated a User/Group schema, created it in SQLite, added a relationship, and checked that the row was stored in `membership_links` and could be queried through the generated edge.

For an already populated database, plan an explicit rename/data migration for the existing join table; changing the schema declaration alone does not express that data-preserving rename.

See the official [StorageKey examples](https://entgo.io/docs/schema-edges/#storagekey).
